### PR TITLE
Remove deployment job from import model pipeline

### PR DESCRIPTION
**Pull request by novaturient95**
**Problem**
Deployment generally has compute quota issues and it's difficult to use it for scale testing.

**Solution**
1. Remove deployment job
2. Use local validation component and have a specific compute provided to it to make sure model can be loaded. This is still an optional step in pipeline, and can be skipped by not passing test data needed for inference

### CODE REVIEW

**Review by HrishikeshGeedMS — APPROVED** [review]

File: training/model_management/components/import_model/asset.yaml [hunk 1]
```diff
@@ -1,2 +1,3 @@
 type: component
 spec: spec.yaml
+categories: ["Import Model"]
```
Comment (skasturi) [inline training/model_management/components/import_model/asset.yaml:3]: Are we listing broad level categories somewhere?

File: training/model_management/components/import_model/spec.yaml [hunk 2]
```diff
@@ -4,14 +4,18 @@ type: pipeline
...
 display_name: Import model
 description: Import a model into a workspace or a registry
-version: 0.0.5
+version: 0.0.6
 
 # Pipeline inputs
 inputs:
   # pipeline specific compute
-  compute:
+  compute_common:
```
Comment (skasturi) [inline training/model_management/components/import_model/spec.yaml:12]: What is the reason to rename this input?

File: training/model_management/components/import_model/spec.yaml [hunk 3]
```diff
@@ -4,14 +4,18 @@ type: pipeline
...
     type: string
-    description: Compute to run pipeline job
+    description: Common compute for model download, mlflow conversion and registration
+
+  compute_local_validation:
```
Comment (skasturi) [inline training/model_management/components/import_model/spec.yaml:16]: Let us get the interface reviewed by @ManojBableshwar

**Review by skasturi — CHANGES_REQUESTED** [review]
Need interface review

File: training/model_management/components/import_model/asset.yaml (hunk 1 above)
Comment (novaturient95) [reply training/model_management/components/import_model/asset.yaml:3]: Hi @elesel could you please help to respond to above query?

Comment (novaturient95) [reply training/model_management/components/import_model/asset.yaml]: Thanks @elesel. Updated categories to `Models`

File: training/model_management/components/import_model/spec.yaml (hunk 2 above)
Comment (novaturient95) [reply training/model_management/components/import_model/spec.yaml]: Reverted

**Review by skasturi — APPROVED** [review]

**Review by vikasagrawal-ms — APPROVED** [review]
Approved

**Pull request merged**